packages/w3up-client/src/client.js

```diff
@@ -282,6 +282,40 @@ export class Client extends Base {
     return space
   }
 
+  /**
+   * Share an existing space with another Storacha account.
+   * Delegates access to the space to the specified email account.
+   *
+   * @typedef {object} ShareOptions
+   * @property {import("./space.js").OwnedSpace} space - The space to share.
+   * @property {import("./types.js").EmailAddress} delegateEmail - Email of the account to share the space with.
+   *
+   * @param {ShareOptions} options
+   * @returns {Promise<void>} Resolves once the space is successfully shared.
+   * @throws {Error} - Throws an error if there is an issue delegating access to the space.
+   */
+  async shareSpace(options) {
```

Comment:
@alanshaw I'm afraid this would require a larger refactor. Apparently, the `Space` type doesn't contain all the methods necessary to be able to execute the delegate flow, e.g:

1. Save the space to create the authorizations (the `Space` type doesn't let you save a space instance)
2. Create a delegation using the access rules you shared (`client.capability.access.delegate` not available at the `Space` type level)
3. Execute the delegate call
4. Check the results and throw an error if it is not OK

Considering we already have the `createSpace` method in the client - which is some sort of space management - I was wondering if it makes sense to keep it within the same context. WDYT?

Reply:
ACK, yes fine by me.